modules/siddhi-extensions/time/src/main/java/org/wso2/siddhi/extension/time/DateAddFunctionExtension.java

```diff
@@ -127,13 +130,12 @@ protected void init(ExpressionExecutor[] attributeExpressionExecutors,
             throw new ExecutionPlanValidationException("Invalid no of arguments passed to time:dateAdd() function, " +
                     "required 3 or 4, but found " + attributeExpressionExecutors.length);
         }
+        unit = (String)((ConstantExpressionExecutor)attributeExpressionExecutors[2]).getValue();
```

Comment:
@dnwick I think it would be better to do an instanceof check first and then do the casting. WDYT? Please see a related discussion for a somewhat similar situation at [1].

[1] https://github.com/wso2/siddhi/pull/158#discussion_r66780639